File: csharp/raindrops/Iteration1/Raindrops.cs
using System;
using System.Collections.Generic;
using System.Linq;

public static class Raindrops
{
    private static readonly Dictionary<int, string> Factors = new()
    {
        {3, "Pling"},
        {5, "Plang"},
        {7, "Plong"}
    };

    public static string Convert(int number)
    {
        return string.Join("",
            Factors
                .Where(p => number % p.Key == 0)
                .Select(p => p.Value)
                .DefaultIfEmpty(number.ToString())
        );
    }
}

File: fsharp/raindrops/Iteration1/Raindrops.fs
module Raindrops

let pling (number: int): string =
    if number % 3 = 0
    then "Pling"
    else ""

let plang (number: int): string =
    if number % 5 = 0
    then "Plang"
    else ""

let plong (number: int): string =
    if number % 7 = 0
    then "Plong"
    else ""

let convert (number: int): string =
    let result =
        [(pling number); (plang number); (plong number)]
        |> String.concat ""

    if System.String.IsNullOrWhiteSpace(result)
    then number.ToString()
    else result